tests/unit/Store.ts

```diff
@@ -0,0 +1,252 @@
+import Store, { StoreArgs, NewableStoreModel } from 'src/Store';
+
+class NonDeclareBasedModel {
+	constructor (args?: {}) {
+		lang.mixin(this, args);
+	}
+	_restore(Constructor: { new (arg: any): { restored: boolean } }) {
+		// use constructor based restoration
+		const restored = new Constructor(this);
+		restored.restored = true;
+		return restored;
+	}
+	static createSubclass: any;
+}
+
+class ConcreteStore<T> extends Store<T> {
+
+	fetchRange(args: dstore.FetchRangeArgs): dstore.FetchPromise<T> {
+		throw new Error('This Method is abstract');
+	}
+
+	add(object: T, directives?: dstore.PutDirectives):  Promise<T> {
+		return Promise.resolve(object);
+	}
+
+	put(object: T, directives?: dstore.PutDirectives):  Promise<T> {
+		return Promise.resolve(object);
+	}
+
+	fetch(args?: dstore.FetchArgs): dstore.FetchPromise<T> {
+		throw new Error('This Method is abstract');
+	}
+
+	get(id: string | number): Promise<T> | void {
+		throw new Error('This Method is abstract');
+	}
+
+	remove(id: any): Promise<T | void> {
+		throw new Error('This Method is abstract');
+	}
+}
+
+let store: ConcreteStore<any>;
+registerSuite({
+	name: 'dstore Store',
+
+	beforeEach: function () {
+		store = new ConcreteStore();
+	},
+
+	'getIdentity and _setIdentity': {
+		'direct property access and assignment'() {
+			const object: Hash<any> = { id: 'default', 'custom-id': 'custom' };
+
+			assert.strictEqual(store.getIdentity(object), 'default');
+			(<any> store)._setIdentity(object, 'assigned-id');
+			assert.strictEqual(object['id'], 'assigned-id');
+			assert.strictEqual(store.getIdentity(object), object['id']);
+
+			store.idProperty = 'custom-id';
+			assert.strictEqual(store.getIdentity(object), 'custom');
+			(<any> store)._setIdentity(object, 'assigned-id');
+			assert.strictEqual(object['custom-id'], 'assigned-id');
+			assert.strictEqual(store.getIdentity(object), object['custom-id']);
+		},
+		'getter and setter'() {
+			const object: Hash<any> = {
+					_properties: {
+						id: 'default',
+						'custom-id': 'custom'
+					},
+					get: function (name: string) {
+						return this._properties[name];
+					},
+					set: function (name: string, value: string) {
+						this._properties[name] = value;
+					}
+				};
+
+			assert.strictEqual(store.getIdentity(object), 'default');
+			(<any> store)._setIdentity(object, 'assigned-id');
+			assert.strictEqual(object['_properties'].id, 'assigned-id');
+			assert.strictEqual(store.getIdentity(object), object['_properties'].id);
+
+			store.idProperty = 'custom-id';
+			assert.strictEqual(store.getIdentity(object), 'custom');
+			(<any> store)._setIdentity(object, 'assigned-id');
+			assert.strictEqual(object['_properties']['custom-id'], 'assigned-id');
+			assert.strictEqual(store.getIdentity(object), object['_properties']['custom-id']);
+		}
+	},
+
+	'filter'() {
+		const filter1 = { prop1: 'one' };
+		const expectedQueryLog1: any[] = [ {
+				type: 'filter', arguments: [ filter1 ], normalizedArguments: [ {
```

Comment:
These lines are indented one level deeper than they should be (through/including `} ];`), same for `expectedQueryLog2`